uri/header_param_encoder.go

```diff
@@ -20,13 +20,26 @@ func (e *headerParamEncoder) serialize() error {
 		e.header.Set(e.paramName, e.val)
 		return nil
 	case typeArray:
-		const sep = ","
-		for _, val := range e.items {
-			if err := checkNotContains(val, sep); err != nil {
-				return err
+		if strings.EqualFold(e.paramName, "set-cookie") {
+			// As per RFC6265:
+			//
+			// Origin servers SHOULD NOT fold multiple Set-Cookie header fields into
+			// a single header field. The usual mechanism for folding HTTP headers
+			// fields (i.e., as defined in RFC2616) might change the semantics of
+			// the Set-Cookie header field because the %x2C (",") character is used
+			// by Set-Cookie in a way that conflicts with such folding.
+			for _, val := range e.items {
+				e.header.Add(e.paramName, val)
+			}
+		} else {
+			const sep = ","
+			for _, val := range e.items {
+				if err := checkNotContains(val, sep); err != nil {
+					return err
+				}
 			}
+			e.header.Set(e.paramName, strings.Join(e.items, sep))
 		}
```

Comment:
Just a little styling issue (avoid else statements if possible).
Usually makes the code more straightforward to read.
```suggestion
		if strings.EqualFold(e.paramName, "set-cookie") {
			// As per RFC6265:
			//
			// Origin servers SHOULD NOT fold multiple Set-Cookie header fields into
			// a single header field. The usual mechanism for folding HTTP headers
			// fields (i.e., as defined in RFC2616) might change the semantics of
			// the Set-Cookie header field because the %x2C (",") character is used
			// by Set-Cookie in a way that conflicts with such folding.
			for _, val := range e.items {
				e.header.Add(e.paramName, val)
			}
			return nil
		}
		
		const sep = ","
		for _, val := range e.items {
			if err := checkNotContains(val, sep); err != nil {
				return err
			}
		}
		e.header.Set(e.paramName, strings.Join(e.items, sep))
```